Providers/Scripts/2.4x-2.5x/Scripts/nxOMSPlugin.py

```diff
@@ -163,6 +172,19 @@ def Set(WorkspaceID, Plugins):
             # log error - neither conf nor plugin directory was found in IP to set
             LG().Log('ERROR', plugin['PluginName'] + ' contains neither plugin nor conf')
             return [-1]
+
+        # Some Plugins (e.g. Security Baseline) have arch specific files
+        if os.path.isdir(plugin_arch_dir):
+            if plugin['Ensure'] == 'Present':
+                # copy all files under plugin
+                copy_all_files(plugin_arch_dir, PLUGIN_PATH, True)
```

Comment:
All uses of copy_all_files with is_exec = True are assuming that all the files in the src path should be made executable. In the future, perhaps there are some architecture-specific files that should not automatically be made executable. This may be the finest level of granularity we can achieve with this code, but I would at least add some documentation to the copy_all_files method to note this shortcoming.